test/pgtest/portals.pt

```diff
@@ -0,0 +1,63 @@
+# Test portal max rows. Note that even though there are two rows returned,
+# even after the second execute a portal suspended is returned, not
+# command complete.
+send
+Query {"query": "BEGIN"}
+Parse {"query": "VALUES (1), (2)"}
+Bind
+Execute {"max_rows": 1}
+Execute {"max_rows": 1}
+Execute {"max_rows": 1}
+Execute {"max_rows": 1}
+Query {"query": "COMMIT"}
+Sync
+----
+
+until
+ReadyForQuery
+ReadyForQuery
+ReadyForQuery
+----
+CommandComplete {"tag":"BEGIN"}
```

Comment:
I know that this isn't part of this pr, and I'm unfamiliar with pgtest/datadriven so not sure if this is something that can change (although it seems like it based on other commits) but I think that it would be clearer if there was an `expect` section with trailing `----` as well. That would also allow starting the the documentation in `lib.rs` with something like "all pgtest sections start with a directive (one of `send`, `until`, or `expect`) and end with four dashes on their own line." Additionally that would allow for verifying that nothing else has been sent, although that seems a bit more difficult/less valuable and would require timeouts or something.

Reply:
I don't understand what this `expect` directive would do. How's it different than `until`? What problem is it solving?

I do think it's valuable to have a "all done" check at the end that verifies that, upon connection termination, there were no unread messages from the server, which would require timeouts or something maybe. Is that what you're talking about?